director/projects/views/mixins.py

```diff
@@ -197,10 +151,9 @@ def get_project(
         return self.project
 
     def get_object(self, *args, **kwargs):
-        self.perform_project_fetch(
+        return self.get_project(
```

Comment:
Changed so we take permissions into account, this allows the `get`, `post` and `delete` methods to be removed as they were mostly there for generic django views.